tools/check-directives.sh

```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 set -e
 
+which perl &>/dev/null || exit 77
```

Comment:
As far as I can remember, I didn't integrate the script into the test suite because, well, nobody seems to look at what is skipped and I kind of want it to fail if it can't be run. For example, currently `perl` isn't installed on Travis CI explicitly so if the debian testing image no longer ships with perl we most likely won't even be aware of that.

Reply:
I think that the test should be written in python3  (which is always available because meson depends on it) so that it could be integrated into the test suite smoothly.

Reply:
We want to catch errors as early as possible. This is the kind of test that, similarly to the test-*-sym and test-tables, is fully deterministic. If it fails anywhere, it'll fail everywhere, and vice-versa. So it's better to run it as part of the test suite on developer machines (where perl is very likely to be installed). Then even if it doesn't run on travis for whatever reason, this doesn't really matter, because it'll get run on ubuntu and in many other places, and having it run in just one place is enough.

Reply:
> We want to catch errors as early as possible.

Can't argue with that and Travis CI is here to catch errors automatically in a predictable manner before they make it to the repository.

> So it's better to run it as part of the test suite on developer machines (where perl is very likely to be installed).

I wouldn't rely on people running the tests manually :-) That's basically why Travis CI exists.

> Then even if it doesn't run on travis for whatever reason, this doesn't really matter, because it'll get run on ubuntu and in many other places, and having it run in just one place is enough.

My point is that it's possible that at some point perl won't be there and the test will be just skipped. I'd have no problem with the PR if it didn't skip the test.

Reply:
Let's ignore the fact that it's very unlikely that perl is not installed on *any* of the CIs that we run, and for the sake of argument assume that we somehow managed to get rid of it.

This test would be skipped. We do skip various tests if dependencies are not available. I don't think this one is so important to make an exception for it.

Reply:
> Let's ignore the fact that it's very unlikely that perl is not installed on any of the CIs that we run, and for the sake of argument assume that we somehow managed to get rid of it.

Instead of assuming the worst, I think it would be much more productive to just install perl explicitly on Travis CI, Ubuntu CI, Centos CI and Semaphore CI.

> This test would be skipped. We do skip various tests if dependencies are not available.

I don't want it to be skipped. That's why I started to run it on Travis CI only where it works now and will fast if perl is no longer there. As I already said, I think that the test should be written in Python to integrate it into the test suite properly.

Reply:
OK. I added `perl` to the package list for travis and semaphore.